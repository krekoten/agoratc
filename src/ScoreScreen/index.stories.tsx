import React from 'react'

import type { Meta, StoryObj } from '@storybook/react'

import ScoreScreen from '.'

const meta: Meta<typeof ScoreScreen> = {
  component: ScoreScreen,
}

export default meta

type Story = StoryObj<typeof ScoreScreen>

export const Layout: Story = {
  render: () => <ScoreScreen score={5} onRestart={() => {}} />,
};
