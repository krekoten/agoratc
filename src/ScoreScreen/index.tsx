import { Box, Button, Typography } from "@mui/material"
import React from "react"

interface ScoreParams {
  score: number,
  onRestart: () => void
}

function ScoreScreen({ score, onRestart }: ScoreParams): JSX.Element {
  return <Box sx={{display: "flex", flexDirection: "column", alignItems: "center", margin: 2}}>
      <Typography variant="h5">Final score</Typography>
      <Typography variant="h6" className="score">{score}</Typography>
      <Button className="restart" variant="contained" color="primary" onClick={onRestart}>Restart</Button>
    </Box>
}

export default ScoreScreen
