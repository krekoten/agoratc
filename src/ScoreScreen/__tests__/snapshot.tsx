import React from "react"
import renderer from "react-test-renderer"

import ScoreScreen from ".."

test("ScoreScreen Snapshot", () => {
  const tree = renderer.create(<ScoreScreen score={5} onRestart={() => {}} />).toJSON()
  expect(tree).toMatchSnapshot()
})
