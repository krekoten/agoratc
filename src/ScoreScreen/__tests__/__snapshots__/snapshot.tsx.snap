// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreScreen Snapshot 1`] = `
<div
  className="MuiBox-root css-ytygsf"
>
  <h5
    className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
  >
    Final score
  </h5>
  <h6
    className="MuiTypography-root MuiTypography-h6 score css-2ulfj5-MuiTypography-root"
  >
    5
  </h6>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium restart css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Restart
  </button>
</div>
`;
